shell/gpu/gpu_surface_metal_impeller.mm

```diff
@@ -214,40 +214,42 @@
 std::unique_ptr<SurfaceFrame> GPUSurfaceMetalImpeller::AcquireFrameFromMTLTexture(
     const SkISize& frame_size) {
   GPUMTLTextureInfo texture_info = delegate_->GetMTLTexture(frame_size);
-  id<MTLTexture> mtl_texture = (id<MTLTexture>)(texture_info.texture);
-
+  id<MTLTexture> mtl_texture = (__bridge id<MTLTexture>)texture_info.texture;
   if (!mtl_texture) {
     FML_LOG(ERROR) << "Invalid MTLTexture given by the embedder.";
     return nullptr;
   }
 
   if (Settings::kSurfaceDataAccessible) {
-    last_texture_.reset([mtl_texture retain]);
+    last_texture_ = mtl_texture;
   }
 
 #ifdef IMPELLER_DEBUG
   impeller::ContextMTL::Cast(*aiks_context_->GetContext()).GetCaptureManager()->StartCapture();
 #endif  // IMPELLER_DEBUG
 
+  __weak id<MTLTexture> weak_texture = mtl_texture;
   SurfaceFrame::EncodeCallback encode_callback =
       fml::MakeCopyable([disable_partial_repaint = disable_partial_repaint_,  //
                          damage = damage_,
                          aiks_context = aiks_context_,  //
-                         mtl_texture                    //
+                         weak_texture                   //
```

Comment:
will it crash if we pass nil here?

Reply:
We check the texture is non-nil right below and bail out if so.